framework/tlvf/src/include/tlvf/WSC/AttrList.h

```diff
@@ -0,0 +1,139 @@
+/* SPDX-License-Identifier: BSD-2-Clause-Patent
+ *
+ * Copyright (c) 2019 Tomer Eliyahu (Intel Corporation)
+ *
+ * This code is subject to the terms of the BSD+Patent license.
+ * See LICENSE file for more details.
+ */
+
+#ifndef _TLVF_WSC_ATTR_LIST_H_
+#define _TLVF_WSC_ATTR_LIST_H_
+
+#include <memory>
+#include <tlvf/ClassList.h>
+#include <tlvf/WSC/WSC_Attributes.h>
+#include <tlvf/ieee_1905_1/tlvWsc.h>
+#include <tlvf/tlvflogging.h>
+
+namespace WSC {
+
+typedef struct sWscAttrHeader {
+    uint16_t type;
+    uint16_t length;
+    uint8_t *data() { return ((uint8_t *)this + sizeof(*this)); }
+} __attribute__((packed)) sWscAttrHeader;
```

Comment:
I'm not sure what's the behavior of all compilers when packing a struct that also contains methods...

Reply:
It is not compiler related AFAIK - the only difference between a struct and class in C++ is the default accessibility of member variables and methods. In a struct they are public; in a class they are private, so there shouldn't be any dependency on one compiler or the other.

Reply:
Right, but you don't use classes the same way you use structs.
For example, you wouldn't `memcpy` into a class or "pack" its' members...
What I meant is that I don't know how the compiler will "pack" the members, since it also contains a method.

Reply:
We can count on the CI to catch it if there is an issue with it..